### DISCUSSION TITLE
can we add release notes according the flavor in release.json

### DISCUSSION THREAD

**Discussion by sailes-coding** [codemagic-ci-cd/codemagic-docs#2123] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Ia have multiple flavorsa in my app so I wan to deploy it in App Store and playstore from Codemagic. Show it is possible to add a release note for each flavor different.  My different flavors have different languages on app store and play store  too

### Expected results

deploying with what new from release json all storeis 

### Actual results

i created the release.json and it whatsnew and language but my flavor doesn't have an English language at all.  so its getting error

**Comment by himesh-cm** [comment]
Hi @sailes-coding, It is possible to generate the `release notes` file dynamically during your workflow and store it in the root directory of your project, according to your specific needs. If you provide me with the white-label workflow that you have configured on Codemagic, I can offer you further guidance on the best practices to follow for your particular use case.

Please share a build ID, as well.

**Comment by sailes-coding** [comment]
will it be ok ??



build id:652e8df52a123a50235d6e26




workflow

# Automatically generated on 2023-02-10 UTC from https://codemagic.io/app/639bdca38fef01003b541f54/settings
# Note that this configuration is not an exact match to UI settings. Review and adjust as necessary.

workflows:
  default-workflow:
    name: Default Workflow
    max_build_duration: 60
    environment:
      groups:
        - enabley
      vars:
        APP_STORE_CONNECT_KEY_IDENTIFIER:
        APP_STORE_CONNECT_PRIVATE_KEY: 
        APP_STORE_CONNECT_ISSUER_ID: 
        FCI_KEYSTORE_PATH: /tmp/keystore.keystore
        FCI_KEYSTORE: 
        FCI_KEYSTORE_PASSWORD: 
        FCI_KEY_PASSWORD: 
        FCI_KEY_ALIAS: 
        CERTIFICATE_PRIVATE_KEY: 
      flutter: stable
      xcode: latest
      cocoapods: default
    triggering:
      events:
        - push
        - pull_request
        - tag
      branch_patterns:
        - pattern: '*'
          include: true
          source: true
      tag_patterns:
        - pattern: '*-prod'
          include: true
    scripts:
      - |
        # set up key.properties
        echo $FCI_KEYSTORE | base64 --decode > $FCI_KEYSTORE_PATH
        cat >> "$FCI_BUILD_DIR/android/key.properties" <<EOF
        storePassword=$FCI_KEYSTORE_PASSWORD
        keyPassword=$FCI_KEY_PASSWORD
        keyAlias=$FCI_KEY_ALIAS
        storeFile=/tmp/keystore.keystore
        EOF
      - |
        # set up local properties
        echo "flutter.sdk=$HOME/programs/flutter" > "$FCI_BUILD_DIR/android/local.properties"
      - flutter packages pub get
      - flutter build appbundle --release --flavor production -t lib/main_production.dart
      - find . -name "Podfile" -execdir pod install \;
      - keychain initialize
      - app-store-connect fetch-signing-files "io.enabley.ping" --type IOS_APP_STORE
        --create
      - keychain add-certificates
      - xcode-project use-profiles
      - flutter build ipa --export-options-plist ~/export_options.plist --flavor production
        -t lib/main_production.dart
    artifacts:
      - build/**/outputs/apk/**/*.apk
      - build/**/outputs/bundle/**/*.aab
      - build/**/outputs/**/mapping.txt
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - '*.snap'
      - build/windows/**/*.msix
      - flutter_drive.log
    publishing:
      email:
    
      app_store_connect:
        api_key: 
        key_id:
        issuer_id: 
        submit_to_testflight: true
        submit_to_app_store: true
  taglit:
    name: taglit
    max_build_duration: 60
    environment:
      groups:
        - enabley
      vars:
        FCI_KEYSTORE_PATH: /tmp/keystore.keystore
        FCI_KEYSTORE: 
        FCI_KEYSTORE_PASSWORD: 
        FCI_KEY_PASSWORD: 
        FCI_KEY_ALIAS: 
      flutter: stable
      xcode: latest
      cocoapods: default
    triggering:
      events:
        - tag
      branch_patterns:
        - pattern: '*'
          include: true
          source: true
      tag_patterns:
        - pattern: '*-prod'
          include: true
    scripts:
      - |
        # set up key.properties
        echo $FCI_KEYSTORE | base64 --decode > $FCI_KEYSTORE_PATH
        cat >> "$FCI_BUILD_DIR/android/key.properties" <<EOF
        storePassword=$FCI_KEYSTORE_PASSWORD
        keyPassword=$FCI_KEY_PASSWORD
        keyAlias=$FCI_KEY_ALIAS
        storeFile=/tmp/keystore.keystore
        EOF
      - |
        # set up local properties
        echo "flutter.sdk=$HOME/programs/flutter" > "$FCI_BUILD_DIR/android/local.properties"
      - flutter packages pub get
      - flutter build appbundle --release --flavor taglit -t lib/main_taglit.dart
        --build-name=1.0.$version_number --build-number=$build_number
    artifacts:
      - build/**/outputs/apk/**/*.apk
      - build/**/outputs/bundle/**/*.aab
      - build/**/outputs/**/mapping.txt
      - '*.snap'
      - build/windows/**/*.msix
      - flutter_drive.log
    publishing:
      email:
        recipients:
          - sailesh.shakya@codingmountain.com
          - df0db946.enabley.io@emea.teams.ms
      google_play:
        credentials: 
        track: production
        in_app_update_priority: 0
      app_store_connect:
        api_key: 
        key_id:
        issuer_id: 
        submit_to_testflight: true
        submit_to_app_store: true
  vivo:
    name: vivo
    max_build_duration: 60
    environment:
      vars:
        FCI_KEYSTORE_PATH: /tmp/keystore.keystore
        FCI_KEYSTORE: 
        FCI_KEYSTORE_PASSWORD: 
        FCI_KEY_PASSWORD: 
        FCI_KEY_ALIAS: 
      flutter: stable
      xcode: latest
      cocoapods: default
    scripts:
      - |
        # set up key.properties
        echo $FCI_KEYSTORE | base64 --decode > $FCI_KEYSTORE_PATH
        cat >> "$FCI_BUILD_DIR/android/key.properties" <<EOF
        storePassword=$FCI_KEYSTORE_PASSWORD
        keyPassword=$FCI_KEY_PASSWORD
        keyAlias=$FCI_KEY_ALIAS
        storeFile=/tmp/keystore.keystore
        EOF
      - |
        # set up local properties
        echo "flutter.sdk=$HOME/programs/flutter" > "$FCI_BUILD_DIR/android/local.properties"
      - flutter packages pub get
      - flutter build appbundle --release --flavor vivo_white_label -t lib/main_vivo_white_label.dart
    artifacts:
      - build/**/outputs/apk/**/*.apk
      - build/**/outputs/bundle/**/*.aab
      - build/**/outputs/**/mapping.txt
      - '*.snap'
      - build/windows/**/*.msix
      - flutter_drive.log
    publishing:
      email:
        recipients:
      google_play:
        credentials: 
        track: production
        in_app_update_priority: 0
      app_store_connect:
        api_key: 
        key_id: 
        issuer_id: 
        submit_to_testflight: true
        submit_to_app_store: true
  comex:
    name: comex
    max_build_duration: 60
    environment:
      vars:
        FCI_KEYSTORE_PATH: /tmp/keystore.keystore
        FCI_KEYSTORE: 
        FCI_KEYSTORE_PASSWORD: 
        FCI_KEY_PASSWORD: 
        FCI_KEY_ALIAS: 
      flutter: stable
      xcode: latest
      cocoapods: default
    scripts:
      - |
        # set up key.properties
        echo $FCI_KEYSTORE | base64 --decode > $FCI_KEYSTORE_PATH
        cat >> "$FCI_BUILD_DIR/android/key.properties" <<EOF
        storePassword=$FCI_KEYSTORE_PASSWORD
        keyPassword=$FCI_KEY_PASSWORD
        keyAlias=$FCI_KEY_ALIAS
        storeFile=/tmp/keystore.keystore
        EOF
      - |
        # set up local properties
        echo "flutter.sdk=$HOME/programs/flutter" > "$FCI_BUILD_DIR/android/local.properties"
      - flutter packages pub get
      - flutter build appbundle --release --flavor comex -t lib/main_comex.dart
    artifacts:
      - build/**/outputs/apk/**/*.apk
      - build/**/outputs/bundle/**/*.aab
      - build/**/outputs/**/mapping.txt
      - '*.snap'
      - build/windows/**/*.msix
      - flutter_drive.log
    publishing:
      email:
    
      app_store_connect:
        api_key: 
        key_id: K2AV6MFLTX
        issuer_id:
        submit_to_testflight: true
        submit_to_app_store: true

**Reply by sailes-coding** [reply]
will it be enought @himesh-cm

**Reply by himesh-cm** [reply]
Hi @sailes-coding, please fill out this [form](https://codemagic.io/contact/) and a member of our solutions engineering team will contact you to better understand your use case and provide assistance.

**Reply by himesh-cm** [reply]
@sailes-coding, Are you still looking for assistance with your query?

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
